packages/editor/src/Theme/Colors.tsx

```diff
@@ -0,0 +1,66 @@
+/** @jsx jsx */
+import { jsx, useThemeUI } from 'theme-ui'
+import { ColorPalette } from '@theme-ui/style-guide'
+import ColorPicker from '../ColorPicker'
+import { EditorContext } from '../types'
+
+type ColorsProps = {
+  size?: number
+}
+type ColorPaletteRenderArg = {
+  color: string
+  swatch: React.ReactNode
+  key: string
+  name: string
+}
+
+const Colors = (props: ColorsProps) => {
+  // TODO: I need to cast the context value because TypeScript assumes the default theme-ui ContextValue. Is there a better way to make TypeScript know of the EditorContext than casting the type?
+  const context = useThemeUI() as EditorContext
+  // FIXME: Where to add colorMode type?
+  const mode = context.colorMode
```

Comment:
I guess the type for `colorMode` will be added in scope of converting the `color-mode` package to TypeScript. Should I wait for it to get merged or try to work around it?

Same applies to `initialColorMode` type.

Reply:
That's correct.

Generally, if this is not a part of public API, there's nothing wrong with `as EditorContext` type assertion.
 
I'm not sure if `setTheme` is public. It seems it's not mentioned in the docs?
https://theme-ui.com/use-theme-ui

If we wanted to tell TypeScript that when we import `@theme-ui/editor` ContextValue is extended we could use module augmentation and declaration merging.

```ts
// in @theme-ui/editor
declare module 'theme-ui' {
  export interface ContextValue {
    setTheme: React.Dispatch<Theme>
  }
}
```

similarly to how we add `sx` prop to React components for use with the jsx pragma.

I am really not sure if `setTheme` is public though @jxnblk.

Reply:
Just curious: Does such an augmentation in any TypeScript file add the `setTheme` property to all `ContextValue`s in the codebase or is it scoped somehow?

Reply:
> Just curious: Does such an augmentation in any TypeScript file add the setTheme property to all ContextValues in the codebase or is it scoped somehow?

It's global. i.e. it adds `setTheme` as long as you have `@theme-ui/editor` installed.

It makes sense for `color-modes`, because there's usually ColorModeProvider at the root, but I'm not sure about the editor.

Reply:
I added `any`s to color-mode-related things and a TODO to remove it when color-modes are converted to TypeScript. Let me know if I should do it differently.